From deeeccf913464635c0add8f3a30472d813a6fdb3 Mon Sep 17 00:00:00 2001
From: Tony Cheneau <tony.cheneau@amnesiak.org>
Date: Mon, 3 Sep 2012 23:26:27 -0400
Subject: [PATCH 12/16] 6lowpan: make memory allocation atomic during 6lowpan
 header creation

This is prevent various crashes when using the serial driver (not yet in
the tree).

Signed-off-by: Tony Cheneau <tony.cheneau@amnesiak.org>
---
 net/ieee802154/6lowpan.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/net/ieee802154/6lowpan.c b/net/ieee802154/6lowpan.c
index ed01c39..bf6548e 100644
--- a/net/ieee802154/6lowpan.c
+++ b/net/ieee802154/6lowpan.c
@@ -396,7 +396,7 @@ static int lowpan_header_create(struct sk_buff *skb,
 		/* TODO:
 		 * if this package isn't ipv6 one, where should it be routed?
 		 */
-	head = kzalloc(100, GFP_KERNEL);
+	head = kzalloc(100, GFP_ATOMIC);
 	if (head == NULL)
 		return -ENOMEM;
 
-- 
1.7.7.6

